### PR TITLE
[ipgen,rstmgr] Generate rstmgr with ipgen

### PR DESCRIPTION
**Pull request by matutem**
This has 18 commits for ease of reviewing. It is a bit challenging since many files are changed, though most of the SV files under ip/rstmgr are just moved to ip_templates/rstmgr.

The individual commits group the changes for ease of reviewing.

Some salient commit changes that may not be obvious:
- [[ipgen,rstmgr] Remove files that are generated](https://github.com/lowRISC/opentitan/pull/21041/commits/821f945a69e9ccba6458871f83fc926b498f0017) wipes out from ip_templates the files that are generated, so the tree will no longer have these useless and confusing files.
- [[ipgen,rstmgr] Turn a few more files into templates and fix paths](https://github.com/lowRISC/opentitan/pull/21041/commits/f764746e0f192b601ce88c46ac10f7294c096def) makes additional files into templates since their generated text contains the name of top.
- [[ipgen,rstmgr] Change templates to use plain dictionaries](https://github.com/lowRISC/opentitan/pull/21041/commits/8570ab68a37281d6a05e5570250959581e3a1f8c) is needed since ipgen uses plain dictionaries, whereas the old flow uses some class objects.
- [[ipgen,rstmgr] Fix BUILD files](https://github.com/lowRISC/opentitan/pull/21041/commits/7b0ed71d2c1c8b2a5a00c1166a20f644c1bf5cb5) adds //hw/ip_templates:all_files to make sure azure sw_build doesn't get into these targets.
- [[ipgen,rstmgr] Fix exclusions for azure sw_build](https://github.com/lowRISC/opentitan/pull/21041/commits/adf6a918126bdd3caa5ec26ca64283ee285b5aff) explicitly instructs azure to stay away from building <ip>_regs.h from ip_templates, since they are useless and we removed the non-generated hjson file.
- [[ipgen,rstmgr] Fix core files](https://github.com/lowRISC/opentitan/pull/21041/commits/70295b0ab080846093c5760c3e0cf4cd3453c5cb) implements the virtual and concrete cores using the `instance_vlnv` function.

The most obvious changes are:
- [[ipgen,rstmgr] Generate rstmgr ip_autogen files with topgen](https://github.com/lowRISC/opentitan/pull/21041/commits/cf7ab5175fb2766e509ee483d41f12eea99d66e0) changes topgen.py to use the ipgen flow. It generates lots of files, but they are all copies or template expansions.
- [[ipgen,rstmgr] Remove hw/ip/rstmgr and old autogen files](https://github.com/lowRISC/opentitan/pull/21041/commits/4f7d6c8a979becd2595126f140505ac4ecc6c4b1) removes all files under hw/ip/rstmgr.

You will realize that other than the changes highlighted above there are lots of path adjustments.

This fixes the rstmgr part of #8440.

### CODE REVIEW

**Comment by matutem** [comment]
I am still dealing with CI issues.

**Comment by a-will** [comment]
CHANGE AUTHORIZED: BLOCKFILE

**Comment by matutem** [comment]
CHANGE AUTHORIZED: BLOCKFILE

File: hw/top_earlgrey/ip_autogen/rstmgr/rstmgr.core [hunk 1]
```diff
@@ -17,17 +19,15 @@ filesets:
       - lowrisc:prim:mubi
       - lowrisc:prim:clock_buf
       - lowrisc:prim:sparse_fsm
-      - "fileset_ip     ? (lowrisc:ip:rstmgr_pkg)"
-      - "fileset_top    ? (lowrisc:systems:rstmgr_pkg)"
+      - lowrisc:opentitan:top_earlgrey_rstmgr_pkg:0.1
+      - lowrisc:opentitan:top_earlgrey_rstmgr_reg:0.1
       - lowrisc:ip:rstmgr_cnsty_chk
-      - "fileset_top    ? (lowrisc:systems:rstmgr)"
-      - "fileset_topgen ? (lowrisc:systems:topgen)"
```
Comment (msfschaffner) [inline hw/top_earlgrey/ip_autogen/rstmgr/rstmgr.core:24]: this is so much nicer than the switches :)

**Review by vogelpi — COMMENTED** [review]
Many thanks for doing all this @matutem ! This conversion is a huge task and the existence of Englishbreakfast doesn't make it any simpler. I am thankful that you manage to do it in a way keeping Englishbreakfast alive as we really need this for the SCA work.

**Pull request merged**